tools/nl/analysis/log/mine.py

```diff
@@ -0,0 +1,186 @@
+flags.DEFINE_string('table', 'nl-query', 'BT table')
+flags.DEFINE_integer('past_days', 5, 'Days to go back')
+flags.DEFINE_string('output_csv', '/tmp/nl_query_log.csv', 'Output CSV')
+
+_COL_SESSION = 'session_info'
+_COL_DATA = 'data'
+_COL_FEEDBACK = 'feedback'
+
+
+def _lidx(ctx, val, d, keys):
+  for k1, k2 in keys:
+    items = val.get(k1, [])
+    if d[k2] >= len(items):
+      logging.error(f'Invalid {ctx} {k1} {k2} {d[k2]} {len(items)}')
+      return {}
+    val = items[d[k2]]
+  return val
+
+
+def _midx(val, keys):
+  for i, k in enumerate(keys):
+    if i == len(keys) - 1:
+      val = val.get(k)
+    else:
+      val = val.get(k, {})
+  return val
+
+
+def _write_rows(key, data, csvw):
+  for fb in data[_COL_FEEDBACK]:
+    if 'queryId' in fb:
+      # Query-level feedback
+      item = _lidx(key, data[_COL_SESSION], fb, [('items', 'queryId')])
+      if not item:
+        continue
+      query = item['query']
+
+      ditem = _lidx(key, data, fb, [(_COL_DATA, 'queryId')])
+      if not ditem:
+        continue
+
+      config = {}
+
+    elif 'chartId' in fb:
+      cid = fb['chartId']
+      qitem = _lidx(key, data[_COL_SESSION], cid, [('items', 'queryIdx')])
+      if not qitem:
+        continue
+      query = qitem['query']
+
+      ditem = _lidx(key, data, cid, [(_COL_DATA, 'queryIdx')])
+      if not ditem:
+        continue
+
+      config = ditem['config']
+      config = _lidx(key, config, cid, [('categories', 'categoryIdx'),
+                                        ('blocks', 'blockIdx'),
+                                        ('columns', 'columnIdx'),
+                                        ('tiles', 'tileIdx')])
+      if not config:
+        continue
+
+    sentiment = fb['sentiment']
+
+    # Format xxx_yyyyyyyy#<project>
+    ts = int(key.split('#')[0].split('_')[1])
+    time = datetime.fromtimestamp(ts / 1000000.0).strftime('%Y-%m-%d %H:%M:%S')
+
+    output_row = {
+        'Time':
+            time,
+        'Query':
+            query,
+        'Detection':
+            _midx(ditem, ['debug', 'detection_type']),
+        'Feedback':
+            sentiment,
+        'ChartTitle':
+            config.get('title'),
+        'ChartType':
+            config.get('type'),
+        'ChartVarKeys':  # Just a few keys for sanity check
+            '\n'.join(config.get('statVarKey', [])[:5]),
+        'Index':  # The index type is nested...
+            _midx(ditem, [
+                'debug', 'query_detection_debug_logs', 'query_transformations',
+                'sv_detection_query_index_type'
+            ]),
+        'SessionId':
+            key,
+    }
+    csvw.writerow(output_row)
+
+
+def mine(table, start, csvw):
+  rows = table.read_rows(filter_=row_filters.TimestampRangeFilter(
+      row_filters.TimestampRange(start=start)))
+  for row in rows:
+    key = row.row_key.decode('utf-8')
+    # Format xxx_yyyyyyyy#<project>
+    project = key.split('#')[1].strip()
+    if project != 'datcom-website-autopush':
```

Comment:
make project a constant or have a function to check this (so it's very clear where the log is from)

Reply:
done.